### PR TITLE
Improve covid progression parameters and add tests

### PR DESCRIPTION
**Pull request by roecla**
The name of this branch is outdated. 

- [x] only alive people who are not in intensive care have contacts
- [x] Add tests for `contacts.py`
- [x] Update the duration of the infectiousness
- [x] Uncouple symptoms and infectiousness
- [x] Move "needing ICU" from `cd_symptoms_false` to `cd_needs_icu_true`
- [x] Fix that individuals scheduled to die can recover before death.
- [x] Update notebooks to use `get_simulate_func`.
- [x] Review notebooks.
- [x] Compare parameters to descriptions.

### CODE REVIEW

File: docs/source/reference_guides/epi_params.rst [hunk 1]
```diff
@@ -22,95 +22,213 @@ Sources:
...
+However, the latency period is rarely given in epidemiological reports on COVID-19.
 Instead, scientists and agencies usually report the incubation period, the period from
 infection to the onset of symptoms. A few studies used measurements of virus shedding to
 estimate infectiousness during the course of the disease. When measurements started
 before the onset of symptoms the development of the viral load before symptoms gives us
 an indication of number of days between the onset of infectiousness and symptoms.
 
 The `ECDC (2020-06-24, question 5)
-<https://www.ecdc.europa.eu/en/covid-19/questions-answers>`_ gives a period of 1-2 days
+<https://www.ecdc.europa.eu/en/COVID-19/questions-answers>`_ gives a period of 1-2 days
```
Comment (tobiasraabe) [inline docs/source/reference_guides/epi_params.rst:51]:
```suggestion
<https://www.ecdc.europa.eu/en/covid-19/questions-answers>`_ gives a period of 1-2 days
```

File: docs/source/reference_guides/epi_params.rst [hunk 2]
```diff
@@ -142,30 +268,18 @@ cases.
...
+    :header: "days until recovery", "probability"
 
-    10%, 15
-    30%, 18
-    30%, 22
-    30%, 27
+    15, 10%
+    18, 30%
+    22, 30%
+    27, 30%
```
Comment (tobiasraabe) [inline docs/source/reference_guides/epi_params.rst:276]: In line with the other numbers.
```suggestion
    15, 0.1
    18, 0.3
    22, 0.3
    27, 0.3
```

File: docs/source/reference_guides/epi_params.rst [hunk 3]
```diff
@@ -183,7 +297,7 @@ rely on data by `the US CDC
     therefore use the maximum of the death and ICU rate.
 
 Other sources often only report the proportion of hospitalized cases admitted to ICU.
-According to the collection of the `MIDAS network <https://midasnetwork.us/covid-19/>`_
+According to the collection of the `MIDAS network <https://midasnetwork.us/COVID-19/>`_
```
Comment (tobiasraabe) [inline docs/source/reference_guides/epi_params.rst:300]:
```suggestion
According to the collection of the `MIDAS network <https://midasnetwork.us/covid-19/>`_
```

File: src/sid/covid_epi_params.csv [hunk 4]
```diff
@@ -63,6 +58,7 @@ cd_symptoms_false,80-100,18,0.2139
 cd_symptoms_false,80-100,22,0.2139
 cd_symptoms_false,80-100,27,0.2139
 cd_symptoms_false,80-100,-1,0.287
+
```
Comment (tobiasraabe) [inline src/sid/covid_epi_params.csv:61]:
```suggestion
```

**Review by tobiasraabe — CHANGES_REQUESTED** [review]
Just some notes while reading your changes to estimate how much time it will take me this afternoon.

File: src/sid/covid_epi_params.csv (hunk 4 above)
Comment (roecla) [reply src/sid/covid_epi_params.csv]: Couldn't accept your suggestion here but removed the blank line.

**Review by tobiasraabe — APPROVED** [review]
This is a fantastic PR which adds a lot of tests and makes the model more realistic by enhancing the countdowns. Very good job, @roecla 🎉

**Pull request merged**